src/issuance.rs

```diff
@@ -238,23 +240,28 @@ pub struct Prepared {
     sighash: [u8; 32],
 }
 
+/// A versioned Issuance authorization signature based on BIP 340 Schnorr.
+pub type VerBIP340IssueAuthSig = VersionedSig<IssueAuthSig<ZSASchnorr>>;
+
 /// Marker for an authorized bundle.
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct Signed {
-    signature: IssueAuthSig<ZSASchnorr>,
+    signature: VerBIP340IssueAuthSig,
 }
 
 impl Signed {
-    /// Returns the signature for this authorization.
-    pub fn signature(&self) -> &IssueAuthSig<ZSASchnorr> {
+    /// Returns the versioned signature for this authorization.
+    pub fn signature(&self) -> &VerBIP340IssueAuthSig {
         &self.signature
     }
 
-    /// Constructs a `Signed` from a byte array containing an `IssueAuthSig` in raw bytes.
-    pub fn from_data(data: &[u8]) -> Self {
-        Signed {
-            signature: IssueAuthSig::decode(data).unwrap(),
-        }
+    /// Constructs a `Signed` from
+    /// - one byte array containing a `SighashVersion` in raw bytes, and
+    /// - one byte array containing an `IssueAuthSig` in raw bytes.
+    pub fn from_data(version: &[u8], sig: &[u8]) -> Option<Self> {
```

Comment:
I don't understand from the comment how to use it. there must be a test to demonstrate.

Reply:
I removed this function
I will implement versioned sig in librustzcash and then I will see whether this function is necessary